### PR TITLE
Add redis_config to redis module

### PR DESCRIPTION
**Pull request by tofagerl**

### CODE REVIEW

**Review by sindrel — COMMENTED** [review]
LGTM. Seeing as most existing instances are v4.0 it seems adding activedefrag as a default setting won't be a breaking change. I'm also assuming the performance impact of adding it to all instances by default is a fair tradeoff?

- I suggest we set maxmemory-policy to allkeys-lfu as a default setting as well
- Could you also add the redis_configs variable in the [module docs](https://github.com/entur/terraform/blob/master/modules/redis/README.md)

**Comment by tofagerl** [comment]
@sindrel As I've read it, activedefrag is pretty safe, yes. Citation: https://serverfault.com/questions/971804/are-there-situations-were-activedefrag-should-be-kept-disabled-in-redis-5-with
Adding `maxmemory-policy` and taking a shot at the docs.

File: modules/redis/main.tf [hunk 1]
```diff
@@ -22,8 +22,8 @@ module "memorystore" {
   enable_apis = "${var.enable_apis}"
 
   reserved_ip_range = var.reserved_ip_range
-
   labels = var.labels
+  redis_configs = var.redis_terrconfigs
```
Comment (sindrel) [inline modules/redis/main.tf:26]: Seems the last commit added a sneaky typo. Assuming this should be `redis_configs = var.redis_configs`

File: modules/redis/README.md [hunk 2]
```diff
@@ -19,26 +19,40 @@ Generated Kubernetes Config Map:
...
+| prevent_destroy                  | Prevents the destruction of the bucket                                                                                                                          |  bool  |   false   |    no    |
+| redis_instance_custom_name       | Redis instance name override                                                                                                                                    | string |    n/a    |    no    |
+| enable_apis                      | Flag for enabling redis API in your project                                                                                                                     |  bool  |   false   |    no    |
+| authorized_network_name_override | Redis instance authorized network name override                                                                                                                 | string |    n/a    |    no    |
+| redis_config                     | Configuration parameters passed to the redis instance on creation in Google Cloud                                                                               |  map   | see below |    no    |
```
Comment (sindrel) [inline modules/redis/README.md:39]: Should be `redis_configs`

File: modules/redis/.terraform.lock.hcl [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# This file is maintained automatically by "terraform init".
```
Comment (sindrel) [inline modules/redis/.terraform.lock.hcl:1]: Think we should leave out the lock file

**Comment by sindrel** [comment]
@tofagerl Added a few inline comments. Looks great otherwise! All good on my end.

File: modules/redis/main.tf (hunk 1 above)
Comment (tofagerl) [reply modules/redis/main.tf:26]: Silly computer takes small breaks for coffee while I'm typing...

**Review by sindrel — APPROVED** [review]

**Pull request merged**